lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1154,7 +1154,7 @@ private function getCommitOrder(array $entityChangeSet = null)
 
                 $joinColumns = reset($assoc['joinColumns']);
 
-                $calc->addDependency($targetClass->name, $class->name, (int)empty($joinColumns['nullable']));
+                $calc->addDependency($targetClass->name, $class->name, (int) ! ( ! isset($joinColumns['nullable']) || $joinColumns['nullable'] === true));
```

Comment:
I just fixed the other requested changes, causing the example to be a lot smaller. This proposed change fixes our broken test, but consequently breaks two other tests. I will look a bit further into this when I have some spare time.

@Ocramius Do you agree that there is a logic inconsistency in the the original line in `UnitOfWork.php`?